diff -ru libNBC-1.1.1-orig/configure libNBC-1.1.1/configure
--- libNBC-1.1.1-orig/configure	2012-07-16 03:50:13.000000000 +0000
+++ libNBC-1.1.1/configure	2013-05-23 20:06:45.067010345 +0000
@@ -20045,18 +20045,8 @@
     fi
 
 
-ac_fn_c_check_header_mongrel "$LINENO" "dcmf.h" "ac_cv_header_dcmf_h" "$ac_includes_default"
-if test "x$ac_cv_header_dcmf_h" = x""yes; then :
-
-
-$as_echo "#define HAVE_DCMF 1" >>confdefs.h
-
-  have_dcmf=1
-
-else
   { $as_echo "$as_me:${as_lineno-$LINENO}: DCMF support disabled" >&5
 $as_echo "$as_me: DCMF support disabled" >&6;}
-fi
 
 
 ac_fn_c_check_func "$LINENO" "MPI_Dist_graph_create" "ac_cv_func_MPI_Dist_graph_create"
@@ -22974,11 +22964,6 @@
   cat nbc.h | sed -e "s/define USE_MPI/define USE_OFED/" > $srcdir/nbc.tmp
   mv $srcdir/nbc.tmp $srcdir/nbc.h
   echo "done"
-elif test x$have_dcmf = x1; then
-  echo -n "configuring DCMF as transport interface (patching nbc.h) ... "
-  cat nbc.h | sed -e "s/define USE_MPI/define USE_DCMF/" > $srcdir/nbc.tmp
-  mv $srcdir/nbc.tmp $srcdir/nbc.h
-  echo "done"
 else
   echo -n "configuring MPI as transport interface (patching nbc.h) ... "
   cat nbc.h | sed -e "s/define USE_OFED/define USE_MPI/" > $srcdir/nbc.tmp
